### PR TITLE
Save and load

### PR DESCRIPTION
**Pull request by phila413**
Updates the save, saveAs, and load JButtons to properly work.  The buttons ask the user what/where the user wishes to save and load files.

### CODE REVIEW

**Comment by phila413** [comment]
Side note: It also fixes the parameter bug that was found during the demo.

File: src/main/java/mike/gui/guiHelperMethods.java [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import java.io.FileReader;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
```
Comment (SerbianSensation) [inline src/main/java/mike/gui/guiHelperMethods.java:9]: Sadly won't let me pinpoint the exact location, but line 27: "import gui.View" needs to be "import mike.gui.View" in order to match the other package names and prevent the build from failing.

**Review by SerbianSensation — CHANGES_REQUESTED** [review]

**Comment by SerbianSensation** [comment]
> Side note: It also fixes the parameter bug that was found during the demo.

Which parameter bug? The one where the name was incorrect?

**Comment by phila413** [comment]
> 
> 
> > Side note: It also fixes the parameter bug that was found during the demo.
> 
> Which parameter bug? The one where the name was incorrect?

Yeah.  The one that made only the first parameter created to be shown each time.

File: src/main/java/mike/gui/guiHelperMethods.java (hunk 1 above)
Comment (SerbianSensation) [reply src/main/java/mike/gui/guiHelperMethods.java:9]: Or you can just get rid of it, IntelliJ let me know that it's not even used.

Comment (phila413) [reply src/main/java/mike/gui/guiHelperMethods.java]: Yeah I'm just gonna remove them instead, since they're not necessary.

**Review by SerbianSensation — COMMENTED** [review]
The gui side looks good, but I tried doing load for cli and it seems to no longer work. Used the path ending with .json and tried ending with just the file name. Said failed to parse directory for both. You guys try and lmk if it worked or failed.

**Comment by phila413** [comment]
> 
> 
> The gui side looks good, but I tried doing load for cli and it seems to no longer work. Used the path ending with .json and tried ending with just the file name. Said failed to parse directory for both. You guys try and lmk if it worked or failed.

Thanks that's a good catch!  I just fixed it.  It was because the path variable was sometimes a relative path and sometimes an absolute path, which was confusing save and load.  I made so that no mater what the user gives the path will always turn into an absolute path.

**Review by SerbianSensation — APPROVED** [review]
So I added the fix to the cli load and now everything seems to work so I will approve it. Since I made the last changes it might be best if someone like Ian or John approve since they didn't write code for this branch. Would be good to merge with develop before the end of the sprint so we can have the save/load gui feature.

**Pull request merged**